scripts/Dockerfile

```diff
@@ -0,0 +1,85 @@
+# NOTE: this Dockerfile is primarily used to compile moon's linux binaries.
+# It supports compiling x86_64 and aarch64 binaries for both glibc and musl.
+#
+# Usage: docker buildx build --output type=local,dest=./output --platform linux/amd64,linux/arm64 -f scripts/Dockerfile .
+
+# glibc
+
+FROM rust:1.62.0 AS base-gnu
+RUN cargo install cargo-chef --version ^0.1
+WORKDIR /app
+COPY .cargo .cargo
+COPY rust-toolchain.toml .
+RUN rustup update
+
+FROM base-gnu AS plan-gnu
+COPY . .
```

Comment:
Should this maybe just be the crates folder and Cargo.* files?

Reply:
Updated 👍 

There's also a `.dockerignore` file that's excluding quite a bit from `COPY . .`, but if you expect only the `Cargo*` and `crates` to affect the build, that's much less likely to bust the cache.

One note about `cargo chef prepare`, the trick there is that it will produce a JSON plan file, and then _only_ that plan file is copied to the next stage, so only the contents hash of the plan file affect the cache key, so it's actually okay for the COPY above to invalidate, it won't invalidate the future stage, if that makes sense.

Reply:
Got it, good to know 👍